src/IO/WriteBufferFromEncryptedFile.cpp

```diff
@@ -23,41 +21,17 @@ WriteBufferFromEncryptedFile::WriteBufferFromEncryptedFile(
 
 WriteBufferFromEncryptedFile::~WriteBufferFromEncryptedFile()
 {
-    /// FIXME move final flush into the caller
-    MemoryTracker::LockExceptionInThread lock(VariableContext::Global);
-    finish();
+    finalize();
 }
 
-void WriteBufferFromEncryptedFile::finish()
-{
-    if (finished)
-        return;
-
-    try
-    {
-        finishImpl();
-        out->finalize();
```

Comment:
it's interesting why we call `out->finalize()` twice, in finishImpl and here

Reply:
I thouht it was a typo, but I could be wrong

Reply:
Seems like the second `out->finalize()` do nothing, because it's just `next` or it has bool flag that prevent double finalizing

Reply:
Yes, looks like a typo